examples/vector-search-with-realtime-data/app.py

```diff
@@ -0,0 +1,631 @@
+        with st.spinner("Setting up embedding function..."):
+            st.session_state.embed_func = setup_embedding_function(st.session_state.db)
+
+    # Setup table
+    if st.session_state.table is None:
+        with st.spinner("Setting up products table..."):
+            st.session_state.table = setup_table(
+                st.session_state.db, st.session_state.embed_func
+            )
+
+    # Load user profile
+    if st.session_state.user_profile is None:
+        st.session_state.user_profile = os.getenv("USER_PROFILE", "a user likes sports")
+
+    # Load initial data
+    table = st.session_state.table
+    if table.rows() == 0:
+        with st.spinner("Loading initial products..."):
+            load_initial_data(table)
+
+
+def main():
+    """Main application"""
+
+    # Setup
+    setup()
+
+    # Title
+    st.markdown(
+        '<h2 style="text-align: center;">🛍️ Vector Search with Real-time Data</h2>',
+        unsafe_allow_html=True,
+    )
+    st.markdown(
+        '<p style="text-align: center; color: #666;">Experience TiDB\'s vector search with live product recommendations</p>',
+        unsafe_allow_html=True,
+    )
+
+    # Settings bar
+    with st.expander("⚙️ Settings", expanded=False):
+        col1, col2 = st.columns(2)
+        with col1:
+            user_profile = st.text_input(
+                "User Profile",
+                value=st.session_state.user_profile,
+                help="Describe the user's preferences for personalized recommendations",
+            )
+            if user_profile != st.session_state.user_profile:
+                st.session_state.user_profile = user_profile
+
+        with col2:
+            distance_threshold = st.slider(
+                "Distance Threshold",
+                min_value=0.0,
+                max_value=2.0,
+                value=0.85,
+                step=0.01,
+                help="Maximum distance to consider a match. 0 = no filter. Lower values = stricter matching.",
+            )
+
+        st.info(
+            "💡 **How it works:** The system uses vector embeddings to match products with your profile. "
+            "Lower distance = better match. Set to 0 to see all results without filtering. "
+            f"Current: {'No filtering' if distance_threshold == 0 else f'Only items within {distance_threshold} distance'}"
+        )
+
+    st.divider()
+
+    # Main layout: Left (Mobile UI) and Right (Admin Panel)
+    left_col, right_col = st.columns([1, 1.5])
+
+    with left_col:
+        st.markdown("### 📱 User View (Shopping App)")
+        # Get recommendations
+        recommendations = get_recommendations(
+            st.session_state.table,
+            st.session_state.user_profile,
+            limit=5,
+            distance_threshold=distance_threshold,
+        )
+        render_mobile_ui(recommendations, st.session_state.user_profile)
+
+    with right_col:
+        render_admin_panel(st.session_state.table)
```

Comment:
Would it be more appropriate to use `st.Page`? For example:

<img width="941" height="583" alt="Image" src="https://github.com/user-attachments/assets/a7bd66f7-1a7e-45f4-8b3e-aa918285c093" />

```suggestion
    with left_col:
        st.markdown("### 📱 User View (Shopping App)")
        # Get recommendations
        recommendations = get_recommendations(
            st.session_state.table,
            st.session_state.user_profile,
            limit=5,
            distance_threshold=distance_threshold,
        )
        render_mobile_ui(recommendations, st.session_state.user_profile)

    with right_col:
        render_admin_panel(st.session_state.table)
```